src/core/wallet/services/batchService/batchService.ts

```diff
@@ -22,48 +21,46 @@ export class BatchService {
   }
 
   public async executeBatch () {
-    const batch = new this.web3Service.web3.BatchRequest();
     const initialNonce = await this.web3Service.web3.eth.getTransactionCount(
       this.walletService.publicKey,
       'pending'
     );
 
-    this.batchTransactions.forEach(async (value, index) => {
+    const transactionsPromise = this.batchTransactions.map(async (value, index) => {
       const transaction = await this.utilsService.signTransaction(value[1], value[0], initialNonce + index);
-      this.transactionInProgress.push({
+      this.web3Service.web3.eth.sendSignedTransaction(transaction.rawTransaction);
+      return {
         txHash: transaction.transactionHash,
         txRow: transaction.rawTransaction,
         creationDate: new Date()
-      });
-
-      batch.add(this.web3Service.web3.eth.sendSignedTransaction(transaction.rawTransaction));
+      };
     });
 
+    const transactions = await Promise.all(transactionsPromise);
     this.batchTransactions = [];
-    batch.execute();
-    return this.watchPendingTransaction();
+    return this.watchPendingTransaction(transactions);
   }
 
-  public watchPendingTransaction () {
+  public watchPendingTransaction (transactionInProgress) {
     return new Promise((resolve, reject) => {
```

Comment:
I think this method never resolves and interval is never cleared

Reply:
check line 68 and above